From 64e62c659649593239ed39ed40a3ae689bf2d593 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 29 Aug 2022 19:53:28 -0700
Subject: [PATCH] Add missing prototypes to function declarations

With Clang 15+ compiler -Wstrict-prototypes is triggering warnings which
are turned into errors with -Werror, this fixes the problem by adding
missing prototypes

Fixes errors like
| log.c:134:24: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
| static void syslog_init()
|                        ^
|                         void

Upstream-Status: Submitted [https://lists.samba.org/archive/rsync/2022-August/032858.html]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Archana Polampalli <archana.polampalli@windriver.com>
---
 checksum.c | 2 +-
 exclude.c  | 2 +-
 log.c      | 2 +-
 main.c     | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/checksum.c b/checksum.c
index 4c91c2b..b3635a0 100644
--- a/checksum.c
+++ b/checksum.c
@@ -796,7 +796,7 @@ static void verify_digest(struct name_num_item *nni, BOOL check_auth_list)
 }
 #endif
 
-void init_checksum_choices()
+void init_checksum_choices(void)
 {
 #if defined SUPPORT_XXH3 || defined USE_OPENSSL
 	struct name_num_item *nni;
diff --git a/exclude.c b/exclude.c
index 7476f6a..b939a0c 100644
--- a/exclude.c
+++ b/exclude.c
@@ -460,7 +460,7 @@ void implied_include_partial_string(const char *s_start, const char *s_end)
 	memcpy(partial_string_buf, s_start, partial_string_len);
 }
 
-void free_implied_include_partial_string()
+void free_implied_include_partial_string(void)
 {
 	if (partial_string_buf) {
 		if (partial_string_len)
diff --git a/log.c b/log.c
index 889b57b..19e5e98 100644
--- a/log.c
+++ b/log.c
@@ -140,7 +140,7 @@ static void logit(int priority, const char *buf)
 	}
 }
 
-static void syslog_init()
+static void syslog_init(void)
 {
 	int options = LOG_PID;
 
diff --git a/main.c b/main.c
index 6050d0e..45722c4 100644
--- a/main.c
+++ b/main.c
@@ -250,7 +250,7 @@ void read_del_stats(int f)
 	stats.deleted_files += stats.deleted_specials = read_varint_bounded(f, 0, MAX_WIRE_DEL_STAT, "deleted_specials");
 }
 
-static void become_copy_as_user()
+static void become_copy_as_user(void)
 {
 	char *gname;
 	uid_t uid;
